nilearn/image/resampling.py

```diff
@@ -185,7 +186,7 @@ def get_mask_bounds(img):
 
     """
     img = _utils.check_niimg(img)
-    mask = img.get_data()
+    mask = (img.get_data() != 0.)
```

Comment:
Memory consumption is the same (we need a boolean copy of img.get_data()). From my experience, direct operations are faster than `astype`.